packages/tpetra/core/test/CrsGraph/CrsGraph_StaticImportExport.cpp

```diff
@@ -208,8 +208,112 @@ TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(CrsGraph, ImportToStaticGraph, LO, GO, NT)
   }
 }
 
+
+TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(CrsGraph, ImportToStaticGraphLocal, LO, GO, NT)
+{
+  // Set up Tpetra typedefs.
+  using map_type = Tpetra::Map<LO, GO, NT>;
+  using graph_type = Tpetra::CrsGraph<LO, GO, NT>;
+  using device_type = typename NT::device_type;
+  using execution_space = typename device_type::execution_space;
+  const char prefix[] = "ImportToStaticGraphLocal: ";
+
+  Tpetra::Details::Behavior::disable_verbose_behavior();
+
+  auto comm = getDefaultComm();
+  const auto my_rank = comm->getRank();
+
+  auto loc_success = 1; // to be revised below
+  auto gbl_success = 0; // output argument
+
+  // The global number of rows in the matrix A to create.  We scale
+  // this relative to the number of (MPI) processes, so that no matter
+  // how many MPI processes you run, every process will have 10 rows.
+  const Tpetra::global_size_t num_gbl_inds = 10 * comm->getSize();
+  const GO idx_base = 0;
+
+  // Construct a Map that is global, but puts all the equations on MPI Proc 0.
+  RCP<const map_type> proc_zero_map;
+  {
+    const size_t num_loc_inds = (my_rank == 0) ? num_gbl_inds : 0;
+    proc_zero_map = rcp(new map_type(num_gbl_inds, num_loc_inds, idx_base, comm));
+  }
+
+  // Create a sparse graph using procZeroMap.
+  auto A = generate_crs_graph<LO,GO,NT>(proc_zero_map);
+  comm->barrier();
+
+  // We've created a sparse matrix that lives entirely on Process 0.
+  // Now we want to distribute it over all the processes.
+  //
+  // Redistribute the matrix.  Since both the source and target Maps
+  // are one-to-one, we could use either an Import or an Export.  If
+  // only the source Map were one-to-one, we would have to use an
+  // Import; if only the target Map were one-to-one, we would have to
+  // use an Export.  We do not allow redistribution using Import or
+  // Export if neither source nor target Map is one-to-one.
+  RCP<graph_type> B;
+  {
+    using export_type = Tpetra::Export<LO,GO,NT>;
+    export_type exporter(proc_zero_map, proc_zero_map);
+    comm->barrier();
+
+    Tpetra::Details::Behavior::enable_verbose_behavior();
+    // Make a new sparse graph whose row map is the global Map.
+    out << prefix << "Creating empty graph from global map.\n";
+    B = rcp(new graph_type(proc_zero_map, 0, Tpetra::StaticProfile));
+    out << prefix << "Empty graph from proc_zero_map map created.\n";
+
+    // Redistribute the data, NOT in place, from graph A (which lives
+    // entirely on Proc 0) to graph B (which is distributed evenly over
+    // the processes).
+    out << prefix << "Performing export operation\n";
+    B->doExport(*A, exporter, Tpetra::INSERT);
```

Comment:
Sierra SD actually reported that this threw an exception.  I had them wrap the doExport call in a try-catch, and used an all-reduce to figure out whether doExport threw.

Reply:
Our failing test within Sierra/SD is fairly small - is there a way for us to output the graph and maps so that we can either send them to you or construct a simpler test that exhibits the same behavior? I haven't looked at your code, but it seems that you are using only a single MPI rank, whereas our example had two.

Reply:
@shardest This example runs with 1-4 MPI processes.  (You have to look at the CMakeLists.txt file in this directory to see that.  The test source code doesn't usually specify the required number of processes.)

Reply:
@shardest wrote:
> ... [i]s there a way for us to output the graph and maps so that we can either send them to you or construct a simpler test that exhibits the same behavior?

Yes -- either way is good, but if you include `MatrixMarket_Tpetra.hpp`, there are a bunch of writers in there, including for Map and CrsGraph.  Thanks!